apps/auxo-app/src/store/vault/vault.thunks.ts

```diff
@@ -1,142 +1,126 @@
 import { createAsyncThunk } from "@reduxjs/toolkit";
-import { Erc20, MerkleAuth, Mono } from "../../types/artifacts/abi";
-import { LibraryProvider } from "../../types/utilities";
+import { Erc20, MerkleAuth, Vault as Auxo } from "../../types/artifacts/abi";
 import { getProof } from "../../utils/merkleProof";
 import { Balance, VaultState } from "./Vault";
 
 /**
- * Thunks allow us to execute asynchronous actions while maintaining a predictable state. 
+ * Thunks allow us to execute asynchronous actions while maintaining a predictable state.
  * In particular, the `createAsyncThunk` method exposes lifecycle methods
  * that can trigger reducers in various parts of the store, depending on the status
  * of a promise.
- * 
+ *
  * Define contract interactions inside the thunk, then return a value that will
  * Be handled by the 'fulfilled' hook, as an extra reducer in the vault slice.
- * 
+ *
  * Rejected values will be passed to the `rejected` handlers.
  * This includes unhandled promise rejections from contract interactions so avoids the necessity
  * for a try/catch block inside the thunk.
- * 
- * If you want notifications to be shown to the user, use the `addTxNotifications` 
- * in the application slice to automatically subscribe to the lifecycle methods 
+ *
+ * If you want notifications to be shown to the user, use the `addTxNotifications`
+ * in the application slice to automatically subscribe to the lifecycle methods
  */
 
-
 /**
- * For ERC20 tokens without a 'permit' method, request an approval from the user for 
+ * For ERC20 tokens without a 'permit' method, request an approval from the user for
  * A transfer of tokens to the auxo vault.
  */
 export type ThunkApproveDepositProps = {
-    deposit: Balance,
-    provider: LibraryProvider,
-    token: Erc20 | undefined,
+  deposit: Balance;
+  token: Erc20 | undefined;
 };
 export const thunkApproveDeposit = createAsyncThunk(
-    'vault/approveDeposit', 
-    async ({
-      deposit, 
-      provider,
-      token
-    }: ThunkApproveDepositProps,
+  "vault/approveDeposit",
+  async (
+    { deposit, token }: ThunkApproveDepositProps,
     { getState, rejectWithValue }
-) => {
+  ) => {
     const { vault } = getState() as { vault: VaultState };
-    if (!token || !vault.selected) return rejectWithValue('Missing token or selected vault');
+    if (!token || !vault.selected)
+      return rejectWithValue("Missing token or selected vault");
     const tx = await token.approve(vault.selected, deposit.value);
```

Comment:
watch out for all these promises where a potential failure is not handled.
My suggestion is that you have a wrapper call of some sort, catching exceptions and managing the faulty state and feedback to the user.

Reply:
createAsyncThunk actually handles this for us without needing to write the try/catch explicitly.

It creates a subscription to the store for the duration of the promise, for any logic contained inside the thunk.

The subscription has 3 lifecycle states:

- pending
- success
- fail

For each contract interaction, I've attached an event listener in the `app.slice` file, if the promise is  rejected or accepted, this will propagate as a notification to the user in the application.

`apps/auxo-app/src/store/app/app.slice.ts`

```ts
  extraReducers: (builder) => {  
    addTxNotifications(builder, thunkMakeDeposit);
    addTxNotifications(builder, thunkApproveDeposit);
    addTxNotifications(builder, thunkConfirmWithdrawal);
    addTxNotifications(builder, thunkIncreaseWithdrawal);
    addTxNotifications(builder, thunkAuthorizeDepositor);
  },
```

I am more than happy to explain in more detail how this works because it's a really neat (IMO) feature of RTK for this kind of problem, that keeps our components & store synchronous and perfomant, while dealing with the async nature of the onChain interactions